validators/code.validator.js

```diff
@@ -1,55 +1,69 @@
-            then: Joi.required(),
-            otherwise: Joi.optional(),
-        }),
-        userAnswer: Joi.string().when('language', {
-            is: [PROMPTV1, PROMPTV2],
-            then: Joi.required(),
-            otherwise: Joi.forbidden(),
-        }),
-        rubric: Joi.string().when('language', {
-            is: [PROMPTV1, PROMPTV2],
-            then: Joi.optional(),
-            otherwise: Joi.forbidden(),
-        }),
-        script: Joi.string()
-            .when('language', {
-                is: [PROMPTV1, PROMPTV2, MULTIFILE],
-                then: Joi.optional(),
-                otherwise: Joi.required(),
-            }),
-        url: Joi.string().trim()
-            .when('language', {
-                is: MULTIFILE,
-                then: Joi.required(),
-                otherwise: Joi.forbidden(),
-            }),
-        type: Joi.string().trim()
+        question: Joi.string().required(),
+        userAnswer: Joi.string().required(),
+        rubric: Joi.string().optional(),
+        script: Joi.string().optional(),
+        points: Joi.number().integer().optional(),
+        hasInputFiles: Joi.bool(),
+        args: Joi.string(),
+        stdin: Joi.string(),
+    })
+}
+
+const _getMultiFileLanguageSchema = () => {
+    return Joi.object({
+        language: Joi.string().required(),
+        question: Joi.string().optional(),
+        script: Joi.string().optional(),
+        url: Joi.string().trim().required(),
+        type: Joi.string()
+            .trim()
             .valid(FRONTEND_REACT_JASMINE, FRONTEND_STATIC_JASMINE)
-            .when('language', {
-                is: MULTIFILE,
-                then: Joi.required(),
-                otherwise: Joi.forbidden(),
-            }),
-        non_editable_files: Joi.object().pattern(
-            Joi.string(),
-            Joi.string().pattern(/^[a-fA-F0-9]{64}$/)
-        ).optional(),
-        points: Joi.number().integer().optional(), // totalScore
+            .required(),
+        non_editable_files: Joi.object()
+            .pattern(Joi.string(), Joi.string().pattern(/^[a-fA-F0-9]{64}$/))
+            .optional(),
+        points: Joi.number().integer().optional(),
         hasInputFiles: Joi.bool(),
         args: Joi.string(),
         stdin: Joi.string(),
     })
+}
+
+const _getDefaultLanguageSchema = () => {
+    return Joi.object({
+        language: Joi.string().required(),
+        question: Joi.string().optional(),
+        script: Joi.string().required(),
+        points: Joi.number().integer().optional(),
+        hasInputFiles: Joi.bool(),
+        args: Joi.string(),
+        stdin: Joi.string(),
+    })
+}
+
+const _getSchemaForLanguage = (language) => {
+    switch (language) {
+    case PROMPTV1:
+    case PROMPTV2:
+    case PROMPTV3:
+        return _getAiLanguageSchema()
+    case MULTIFILE:
+        return _getMultiFileLanguageSchema()
+    default:
+        return _getDefaultLanguageSchema()
+    }
```

Comment:
Just verify that DB type validation is getting correctly handled in the default case.

Reply:
Yeah will verify this flow